src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,210 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public boolean add(T value) {
-        return false;
+        linkLast(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkElementIndex(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        return addAll(size, list);
+    }
+
+    private boolean addAll(int index, List<T> c) {
+        checkElementIndex(index);
+        Object[] a = c.toArray();
+        int lengthList = a.length;
+        if (lengthList == 0) {
+            return false;
+        }
+
+        Node<T> pred;
+        Node<T> succ;
+
+        if (index == size) {
+            succ = null;
+            pred = last;
+        } else {
+            succ = node(index);
+            pred = succ.prev;
+        }
+        for (Object o : a) {
+            T e = (T) o;
+            Node<T> newNode = new Node<>(pred, e, null);
+            if (pred == null) {
+                first = newNode;
+            } else {
+                pred.next = newNode;
+            }
+            pred = newNode;
+        }
+        if (succ == null) {
+            last = pred;
+        } else {
+            pred.next = succ;
+            succ.prev = pred;
+        }
+        size += lengthList;
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkPositionIndex(index);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkPositionIndex(index);
+        Node<T> x = node(index);
+        T oldVal = x.item;
+        x.item = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkPositionIndex(index);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T t) {
+        if (t == null) {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
```

Comment:
you have repeating code here, make one check that will be able to work with objects and null at the same time and you will make this method twice shorter

Reply:
Done